test/cpp/ir/core/program_translator_test.cc

```diff
@@ -107,3 +109,39 @@ TEST(RegisterInfoTest, MainProgram) {
   EXPECT_EQ(unregistered_ops.size(), 1u);
   EXPECT_EQ(unregistered_ops[0], "something must not be registered");
 }
+
+TEST(IrParserTest, MainProgram) {
+  auto p = load_from_file("resnet50_main.prog");
+  EXPECT_EQ(p.Size(), 1u);
+  ir::IrContext *ctx = ir::IrContext::Instance();
+  ctx->GetOrRegisterDialect<PaddleDialect>();
+  ctx->GetOrRegisterDialect<ir::BuiltinDialect>();
+  auto program = paddle::TranslateLegacyProgramToProgram(p);
+
+  std::stringstream ss;
+  program->Print(ss);
+  ir::IrParser *parser = new ir::IrParser(ctx, ss);
+  std::unique_ptr<ir::Program> parser_program = parser->ParseProgram();
+  std::stringstream ssp;
+  parser_program->Print(ssp);
+  delete parser;
+  EXPECT_TRUE(ssp.str() == ss.str());
```

Comment:
给Program新增个Parse接口吧
```suggestion
std::ifstream program_text(file_path);
std::unique_ptr<ir::Program> program = ir::Program::Parse(program_text);
```
这样写太麻烦了

Reply:
已添加